### PR TITLE
Quick wins in IDE sync

### PR DESCRIPTION
**Pull request by melix**
This PR collects a number of quick wins following profiler data from IDE synchronization of the `gradle/gradle` build itself.

### CODE REVIEW

File: subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeBuildDependencySubstitutions.java [hunk 1]
```diff
@@ -74,12 +74,16 @@ private ProjectComponentIdentifier getReplacementFor(ModuleComponentSelector sel
         ModuleIdentifier candidateId = selector.getModuleIdentifier();
         Collection<ProjectComponentIdentifier> providingProjects = replacementMap.get(candidateId);
         if (providingProjects.isEmpty()) {
-            LOGGER.debug("Found no composite build substitute for module '" + candidateId + "'.");
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Found no composite build substitute for module '" + candidateId + "'.");
```
Comment (ljacomet) [inline subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeBuildDependencySubstitutions.java]: Is it not simpler to use: `LOGGER.debug("Found no composite build substitute for module '{}'.", candidateId);

Comment (melix) [reply subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeBuildDependencySubstitutions.java]: It is but that's not equivalent though, because _if_ there's logging, the formatter will have to parse the pattern and substitute, which is using something similar to what `String.format` does.

Comment (lptr) [reply subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeBuildDependencySubstitutions.java]: Sorry, this is a pet peeve of mine. We should do what @ljacomet suggests, because:

- parsing the {} is not as slow as `String.format()`, but much faster,
- by adding the `if` now you check the log level twice when debug logging is enabled, which needs to go through synchronization etc, and can actually make this slower than the `if`-less variant,
- performance is irrelevant, since we will only parse the string when debug logging is enabled.

The bottom line is: let's use the simpler approach. The main point is to avoid the string concatenation, and that will work with using `{}`.

Comment (melix) [reply subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeBuildDependencySubstitutions.java]: I changed it. As for whether performance is relevant or not, for the debug case, logging is probably slowing the build already, so no big deal. For the info use case below, that's more a concern, but the situation won't be as bad as it is today.

File: subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeBuildDependencySubstitutions.java [hunk 2]
```diff
@@ -74,12 +74,12 @@ private ProjectComponentIdentifier getReplacementFor(ModuleComponentSelector sel
...
         Collection<ProjectComponentIdentifier> providingProjects = replacementMap.get(candidateId);
         if (providingProjects.isEmpty()) {
-            LOGGER.debug("Found no composite build substitute for module '" + candidateId + "'.");
+            LOGGER.debug("Found no composite build substitute for module '{}'.", candidateId);
             return null;
         }
         if (providingProjects.size() == 1) {
             ProjectComponentIdentifier match = providingProjects.iterator().next();
-            LOGGER.info("Found project '" + match + "' as substitute for module '" + candidateId + "'.");
+                LOGGER.info("Found project '{}' as substitute for module '{}'.", match, candidateId);
```
Comment (ljacomet) [inline subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeBuildDependencySubstitutions.java]: Formatting is off here

**Comment by melix** [comment]
@donat FYI
